webpack/hashed-folder-copy-plugin/src/HashedFolderCopyPlugin.ts

```diff
@@ -0,0 +1,377 @@
+          let dependencyText: string;
+          if (!requireFolderOptions) {
+            dependencyText = this._renderError(errorMessage!);
+            compilation.errors.push(new Error(errorMessage));
+          } else {
+            dependencyText = this._collectAssets({
+              module: currentModule,
+              compilation,
+              requireFolderOptions,
+              resolver: normalResolver
+            });
+          }
+
+          const errorDependency: ConstDependency = new this._ConstDependency(
+            `/* ${EXPRESSION_NAME} */ ${dependencyText}`,
+            expression.range,
+            false
+          );
+          errorDependency.loc = expression.loc;
+          currentModule.addDependency(errorDependency);
+        });
+
+        parser.hooks.evaluateTypeof
+          .for(EXPRESSION_NAME)
+          .tap(PLUGIN_NAME, ParserHelpers.evaluateToString('function'));
+
+        parser.hooks.typeof
+          .for(EXPRESSION_NAME)
+          .tap(PLUGIN_NAME, ParserHelpers.toConstantDependency(parser, JSON.stringify('function')));
+      };
+
+      normalModuleFactory.hooks.parser.for('javascript/auto').tap(PLUGIN_NAME, handler);
+      normalModuleFactory.hooks.parser.for('javascript/dynamic').tap(PLUGIN_NAME, handler);
+    });
+  }
+
+  private _collectAssets(options: ICollectAssetsOptions): string {
+    const { module, compilation, requireFolderOptions, resolver } = options;
+
+    // Map of asset names (to be prepended by the outputFolder) to asset objects
+    const assetsToAdd: Map<string, IAsset> = new Map<string, IAsset>();
+
+    for (const source of requireFolderOptions.sources) {
+      const { globsBase, globPatterns } = source;
+
+      let resolvedGlobsBase: string;
+      if (globsBase.startsWith('.')) {
+        // Does this look like a relative path?
+        resolvedGlobsBase = path.resolve(module.context, globsBase);
+      } else if (path.isAbsolute(globsBase)) {
+        // This is an absolute path
+        resolvedGlobsBase = globsBase;
+      } else {
+        // This looks like a NodeJS module path
+        let slashAfterPackageNameIndex: number;
+        if (globsBase.startsWith('@')) {
+          // The package name has a scope
+          slashAfterPackageNameIndex = globsBase.indexOf('/', globsBase.indexOf('/') + 1);
+        } else {
+          slashAfterPackageNameIndex = globsBase.indexOf('/');
+        }
+
+        let packageName: string;
+        let pathInsidePackage: string;
+        if (slashAfterPackageNameIndex === -1) {
+          packageName = globsBase;
+          pathInsidePackage = '';
+        } else {
+          packageName = globsBase.substr(0, slashAfterPackageNameIndex);
+          pathInsidePackage = globsBase.substr(slashAfterPackageNameIndex + 1);
+        }
+
+        let packagePath: string;
+        try {
+          const resolveResult: string = resolver.resolveSync(
```

Comment:
Can we resolveAsync instead and then catch in the callback?

Reply:
Unfortunately, no. This hook is a `SyncBailHook`.